### PR TITLE
[Mistweaver] Add Expel Harm Module

### PR DESCRIPTION
**Pull request by pingypong**
-added spellId for mistweaver's targeted expel harm healing 
-added ExpelHarm.js which handles healing for both expel harm spellIds and mastery healing
-will be used to to add expel harm to mastery source breakdown stat box

### CODE REVIEW

File: src/parser/monk/mistweaver/modules/spells/ExpelHarm.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import SPELLS from 'common/SPELLS';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Combatants from 'parser/shared/modules/Combatants';
+import Events from 'parser/core/Events';
+import AbilityTracker from 'parser/shared/modules/AbilityTracker';
+
+class ExpelHarm extends Analyzer {
+    static dependencies = {
+        abilityTracker: AbilityTracker,
```
Comment (abelito75) [inline src/parser/monk/mistweaver/modules/spells/ExpelHarm.js:9]: Unneed

File: src/parser/monk/mistweaver/modules/spells/ExpelHarm.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        combatants: Combatants,
```
Comment (abelito75) [inline src/parser/monk/mistweaver/modules/spells/ExpelHarm.js:10]: Unneeded

**Review by abelito75 — CHANGES_REQUESTED** [review]
You might want to add Expel Harm to MistweaverHealingEfficiencyTracker and MasteryStats as well

File: src/parser/monk/mistweaver/modules/spells/ExpelHarm.js (hunk 2 above)
Comment (pingypong) [reply src/parser/monk/mistweaver/modules/spells/ExpelHarm.js:10]: oops, artifacts from previous iteration

**Comment by pingypong** [comment]
mastery and healing efficiency changes were going to be a separate pr based on our previous discussion but I  can toss them in here if you like

**Comment by abelito75** [comment]
I think they apply to this PR. What I was saying before was like `oh i added expel harm now let me go add chi-ji to MistweaverHealingEfficiencyTracker randomly. If that makes sense

File: src/parser/monk/mistweaver/modules/features/MistweaverHealingEfficiencyTracker.js [hunk 3]
```diff
@@ -124,8 +128,11 @@ class MistweaverHealingEfficiencyTracker extends HealingEfficiencyTracker {
     spellInfo.overhealingDone = this.healingDone.byAbility(SPELLS.GUST_OF_MISTS_CHIJI.id).overheal;
     return spellInfo;
   }
+  getExpelHarmDetails(spellInfo) {
+    spellInfo.healingDone = spellInfo.healingDone + this.expelHarm.gustsHealing + this.expelHarm.selfHealing + this.expelHarm.targetHealing;
```
Comment (abelito75) [inline src/parser/monk/mistweaver/modules/features/MistweaverHealingEfficiencyTracker.js:132]: include over healing

**Review by abelito75 — CHANGES_REQUESTED** [review]

File: src/parser/monk/mistweaver/modules/features/MistweaverHealingEfficiencyTracker.js (hunk 3 above)
Comment (pingypong) [reply src/parser/monk/mistweaver/modules/features/MistweaverHealingEfficiencyTracker.js:132]: added

**Review by abelito75 — APPROVED** [review]

**Pull request merged**